src/particles/ChargeDeposition.cpp

```diff
@@ -0,0 +1,99 @@
+/* Copyright 2022 Axel Huebl
+ *
+ * This file is part of ImpactX.
+ *
+ * License: BSD-3-Clause-LBNL
+ */
+#include "ImpactXParticleContainer.H"
+
+#include <ablastr/particles/DepositCharge.H>
+
+#include <AMReX_AmrParGDB.H>
+#include <AMReX_ParallelDescriptor.H>
+#include <AMReX_ParticleTile.H>
+
+#include <array>
+
+
+namespace impactx
+{
+    void
+    ImpactXParticleContainer::DepositCharge (amrex::MultiFab & rho, int icomp)
+    {
+        // loop over refinement levels
+        int const nLevel = this->maxLevel();
+        for (int lev = 0; lev <= nLevel; ++lev) {
```

Comment:
Here it seems that the particles from all levels deposit to the same `MultiFab` `rho`.

This is a bit different than the way we do things in WarpX, in which we:
- Deposit the particles from level 0 into `rho[lev=0]`, particles from level 1 into `rho[lev=1]`, etc.
- Then we use coarsening and `ParallelCopy` in order to propagate the charge deposited in `rho[lev=1]` down to `rho[lev=0]`
The reason we do it this way in WarpX is that the distribution mapping of `rho[lev=0]` and particles in level 0 is related, and the distribution mapping of `rho[lev=1]` and particles in level 1 is related too. This way we avoid particles trying to deposit into cells that are not owned by the local MPI rank.